xbmc/platform/posix/filesystem/PosixFile.cpp

```diff
@@ -331,7 +334,39 @@ int CPosixFile::Stat(const CURL& url, struct __stat64* buffer)
   if (filename.empty() || !buffer)
     return -1;
 
+#if defined(HAS_STATX)
+  int dirfd = AT_FDCWD;
+  int flags = AT_SYMLINK_NOFOLLOW;
+  unsigned int mask = STATX_ALL;
+  struct statx stxbuf = {};
+  long ret = 0;
+  ret = statx(dirfd, filename.c_str(), flags, mask, &stxbuf);
```

Comment:
Can you add a comment that explains why `statx` is used at all?

Reply:
Yes sure.